### PR TITLE
@gridsome/source-datocms plugin

### PR DESCRIPTION
**Pull request by stefanoverna**
Alright @hjvedvik, this works with gridsome version on master! The only thing that's not working is refs with multiple nodes, but I think that's related to this https://github.com/gridsome/gridsome/blob/master/gridsome/lib/graphql/schema/nodes/createType.js#L127-L130

Let me know if everything's fine with this PR!

### CODE REVIEW

File: packages/source-datocms/index.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    for (const itemType of itemsRepo.itemTypes) {
+      const { titleField, fields } = itemType
+
+      const slugField = fields.find(({ fieldType }) => fieldType === 'slug')
+
+      cache[itemType.id] = { titleField, slugField }
+
+      const contentType = {
+        typeName: makeTypeName(itemType.name),
+        route: routes[itemType.name] || `${slugify(itemType.name)}/:slug`,
```
Comment (hjvedvik) [inline packages/source-datocms/index.js]: Can you prefix the fallback route with a `/`?

**Comment by stefanoverna** [comment]
Sure!

Il giorno ven 9 nov 2018, 13:10 Hans-Jørgen Vedvik <notifications@github.com>
ha scritto:

**Comment by hjvedvik** [comment]
You are right, references with multiple node types isn't ready yet, but I want to look into it as soon as possible. I've only added one minor comment to the code that should be fixed, but everything else looks good :)

**Comment by Devportobello** [comment]
Hi @stefanoverna 

Can you show me example of using with "page-query" for "single record" and "multiple records" ?

For single record, i'm facing an "A DatoCmsHome with id undefined was not found" error while no error when using the graphql.datocms.com :/

**Pull request merged**